tests/integration/generic_test.go

```diff
@@ -0,0 +1,176 @@
+package integration
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+
+	"fmt"
+)
+
+var _ = Describe("odo generic", func() {
+	// TODO: A neater way to provide odo path. Currently we assume \
+	// odo and oc in $PATH already.
+	var project string
+	var context string
+	var oc helper.OcRunner
+	var err error
+	var testPHPGitURL = "https://github.com/appuio/example-php-sti-helloworld"
+
+	BeforeEach(func() {
+		oc = helper.NewOcRunner("oc")
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+	})
+
+	Context("creating component without an application and url", func() {
+		JustBeforeEach(func() {
+			project = helper.CreateRandProject()
+		})
+
+		JustAfterEach(func() {
+			helper.DeleteProject(project)
+			os.RemoveAll(".odo")
+		})
+		// https://github.com/openshift/odo/pull/1634
+		/*It("should create the component in default application", func() {
```

Comment:
why is this test commented out, but below we have another git component that is ok?

Reply:
This is commented because push does not work after updating the ```Ports``` in the config. Where as the below component there is no ```Ports``` set in the config. 

Hope fix of #1634 will resolve the issue.

Reply:
Could be related to #1695